### PR TITLE
Add formatting to profile fields for artists

### PR DESCRIPTION
**Pull request by oskarrough**
I configured the API to return HTML for "profile" field.

I did it with a computed property on the model that returns whichever profile field is available, prefering HTML or plaintext to the bbcode.

Other models also need it but I did not feel like creating a shared model or component yet. Thoughts?

Example https://deploy-preview-47--explorer-discogs.netlify.com/artists/15885/info

### CODE REVIEW

**Comment by hugurp** [comment]
Now that I am thinking about it, the best would be to parse the bb code and return the appropriate `explorer` route.

Right now it opens a lot of links to discogs.

**Comment by oskarrough** [comment]
True, I did not consider that. I'd say the easiest would be to write a helper that parses the HTML and replaces `discogs.com/xxx` with our paths?

**Comment by hugurp** [comment]
Updated this one so it has two capture group, `model type` and `model id` https://regexr.com/3pkgo.
Not sure how to remove the tailing slash at the end of the first capture group.

File: app/helpers/format-profile.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import {helper} from '@ember/component/helper'
+
+const hrefRegex = /href="(.*?)"/gm
+const typeAndIdRegex = /([a-z]+(?:\/))([0-9]+(?:$|(?=[?-])|(?=\/$)))/gm
```
Comment (hugurp) [inline app/helpers/format-profile.js]: `typeAndIdFromUrl`?

Comment (hugurp) [reply app/helpers/format-profile.js]: Would also make a nice little `npm` package no? Could be reused for sure in r4

File: app/helpers/format-profile.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export function formatProfile([profile]) {
+  return profile.replace(hrefRegex, (href) => {
```
Comment (hugurp) [inline app/helpers/format-profile.js]: neat!

File: app/helpers/format-profile.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    let [typeAndId] = href.match(typeAndIdRegex)
+    const [type, id] = typeAndId.split('/')
+    return `href="/${type}s/${id}"`
+  })
+}
```
Comment (hugurp) [inline app/helpers/format-profile.js]: neet ^ !

File: app/helpers/format-profile.js (hunk 1 above)
Comment (oskarrough) [reply app/helpers/format-profile.js]: for npm package i'd say let's wait a bit?

here's a very basic test https://github.com/internet4000/explorer-discogs/blob/086d53cc2fc0c7726082ddc8835f7fda803f884d/tests/integration/helpers/format-profile-test.js

**Pull request merged**